ppr-api/src/ppr_api/models/search_request.py

```diff
@@ -225,15 +225,26 @@ def search_by_business_name(self):
             self.returned_results_size = 0
             self.total_results_size = 0
 
-    def search_by_individual_name(self):
+    def search_by_individual_name(self):  # pylint: disable=too-many-locals; easier to follow
         """Execute a debtor individual name search query."""
+        result = None
+        middle_name = None
         last_name = self.request_json['criteria']['debtorName']['last']
         first_name = self.request_json['criteria']['debtorName']['first']
+        if 'second' in self.request_json['criteria']['debtorName']:
+            middle_name = self.request_json['criteria']['debtorName']['second']
         max_results_size = int(current_app.config.get('ACCOUNT_SEARCH_MAX_RESULTS'))
-        result = db.session.execute(search_utils.INDIVIDUAL_NAME_QUERY,
-                                    {'query_last': last_name.strip().upper(),
-                                     'query_first': first_name.strip().upper(),
-                                     'max_results_size': max_results_size})
+        if middle_name is not None and middle_name.strip() != '':
+            result = db.session.execute(search_utils.INDIVIDUAL_NAME_MIDDLE_QUERY,
+                                        {'query_last': last_name.strip().upper(),
+                                         'query_first': first_name.strip().upper(),
+                                         'query_middle': middle_name.strip().upper(),
```

Comment:
will middle name always be populated? I see that you check the field is there, but is it possible it is None and could cause an error here?

Reply:
Thanks Kial, I have added another check in a second commit.